File: codechef/RCSN2015/recrep.java
import java.util.LinkedList;
import java.io.*;
class Main
{
	public static void main(String args[])throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		int t=Integer.parseInt(br.readLine());
		while(t-->0)
		{
			int n=Integer.parseInt(br.readLine());
			LinkedList names=new LinkedList();
			for(int i=0;i<n;i++)
				names.add(br.readLine());
			String name=br.readLine();
			int i;
			for(i=0;i<n;i++)
			{
				if(names.get(i).toString().equals(name))
				{
					System.out.println("yes");
					break;
				}
			}
			if(i==n)
				System.out.println("no");
		}
	}
}

File: codechef/RCSN2015/recrep.c
#include<stdlib.h>
#include<stdio.h>
#include<string.h>
int main()
{
	int t,i,n;
	char names[1001][101],name[101];
	scanf("%d",&t);
	while(t--)
	{
		scanf("%d",&n);
		for(i=0;i<n;i++)
			scanf("%s",names[i]);
		scanf("%s",name);
		for(i=0;i<n;i++)
		{
			if(strcmp(names[i],name)==0)
			{
				printf("yes\n");
				break;
			}
		}
		if(i==n)
			printf("no\n");
	}
	return 0;
}

